### ISSUE TITLE
"index out of bounds" error on "fn z () { <fn ("

### ISSUE THREAD

**Issue by dwrensha** [shepmaster/fuzzy-pickles#57]
```
$ cargo build
   Compiling fuzzy-pickles v0.1.0
    Finished dev [unoptimized + debuginfo] target(s) in 39.7 secs
$ echo "fn z () { <fn (" > foo.rs
$ RUST_BACKTRACE=1 ./target/debug/stats foo.rs
thread 'main' panicked at 'index out of bounds: the len is 8 but the index is 8', src/lib.rs:134
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: fuzzy_pickles::State::ex
             at ./src/lib.rs:134
   1: fuzzy_pickles::function_qualifiers::{{closure}}
             at ./src/lib.rs:2946
   2: fuzzy_pickles::function_qualifiers
             at ./src/lib.rs:2935
   3: fuzzy_pickles::typ_function
             at ./src/lib.rs:5278
   4: core::ops::FnOnce::call_once
             at /checkout/src/libcore/ops.rs:2606
...
  47: stats::main
             at ./src/bin/stats.rs:28
```

**Comment by dwrensha** [comment]
Another example with the same error: `c!(`

**Comment by shepmaster** [comment]
Hmm, a similar type of failure as #53 — we hit the end of the tokens are are trying to report that. In the back of my head I was thinking about introducing an EOF token. Perhaps such a token would help here as well...

**Comment by shepmaster** [comment]
Closed via #58
